internal/secret/store.go

```diff
@@ -115,7 +115,6 @@ func (s *Store) Create(name string, secret Secret) (err error) {
 	if err = s.Remote.Create(name, secret.String()); err != nil {
 		return err
 	}
-	s.cache.SetOrGet(name, secret)
```

Comment:
Is this expected? @aead

Reply:
Yes @harshavardhana - this another improvement against data races. Now, creating a key does not load the created key directly into the cache. Instead, KES fetches the key on it's fist actual usage.

That means, that two concurrent `CreateKey` requests to two KES servers don't affect the caches of those KES servers. Instead, they fetch the created key once it is actually used. Now, the likelihood of two KES servers holding with different keys (with the same name) in the cache becomes significantly smaller.

Reply:
I see @aead is there a likely hood of this happening with kes v0.10 versions

Reply:
They can - but all all other backends, except for Azure KeyVault, support atomic creates. So, if two `CreateKey` calls happen concurrently then one will win while the other will fail. Therefore, it was "cheaper" to also load the key into the cache once creation succeeded. However, this was a micro optimisation and should not really matter w.r.t. performance.

Reply:
The data race issues / concurrency problems are specific to Azure KeyVault.